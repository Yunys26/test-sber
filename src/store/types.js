const GETDATA = 'GETDATA';
const LIKEWORK = 'LIKEWORK';