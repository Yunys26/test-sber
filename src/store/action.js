import { GETDATA, LIKEWORK} from './types';

export function get () {
    return {
        type: 'GETDATA'
    }
}
export function post () {
    return {
        type: 'LIKEWORK'
    }
}
