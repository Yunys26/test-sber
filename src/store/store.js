// Libs
import { configureStore, combineReducers, getDefaultMiddleware } from "@reduxjs/toolkit";
import mainSlice from './slicesStore/mainSlice';

// Создание главного редьюсера
const rootReducer = combineReducers({
    main: mainSlice,
});

// Настройка хранилища
const middleware = getDefaultMiddleware({
    immutableCheck: false,
    serializableCheck: false,
    thunk: true,
});

// Хранилище
export const store = configureStore({
    reducer: rootReducer,
    middleware,
});
