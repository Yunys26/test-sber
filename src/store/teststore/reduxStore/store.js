import { createStore } from 'redux';
import initialState from './initialState';
import { ACTION_1, ACTION_2, ACTION_3, ACTION_4 } from './action/types';

// Создание глобального хранилища
const store = createStore(rootReducer, initialState);

function reducer (state = initialState, action) {
    switch(action.type) {
        case ACTION_1: return { value: action.value_1 };
        case ACTION_2: return { value: action.value_2 };
        case ACTION_3: return { value: action.value_2 };
        case ACTION_4: return { value: action.value_2 };

        default: return state;
    }
}

export default store;