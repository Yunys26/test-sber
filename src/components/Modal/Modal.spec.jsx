import React from 'react';
import { unmountComponentAtNode } from "react-dom";
import { shallow } from 'enzyme';
import ModalS from './Modal';
import { Provider } from 'react-redux';
import { store } from '../../store/store';



describe('<Modal />', () => {
    let props, componentModal; 

    beforeEach(() => {

        props = {
            open: false,
        };

        componentModal = shallow(<Provider store={store}><ModalS {...props}/></Provider>)
    });

    it("Проверка на рендер компонента", () => {
        expect(componentModal).toBeTruthy();
    });

    it('нажатие кнопки "Удалить все"', () => {
        componentModal.find('button').simulate('click');
        expect()
    });

    // it('нажатие кнопки "Ок"', () => {
    //     componentModal.find('button').simulate('click');
    //     expect()
    // });
        
    it('render initial', () => {
        expect(componentModal.find('button')) // .find + поиск по имени компонента
        expect(componentModal.find('h2'))
    });

});
