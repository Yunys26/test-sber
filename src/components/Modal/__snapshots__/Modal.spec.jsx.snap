// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Содержимое <Modal />  Snapshot Modal 1`] = `
<ForwardRef(Modal)
  BackdropProps={
    Object {
      "timeout": 500,
    }
  }
  aria-describedby="spring-modal-description"
  aria-labelledby="spring-modal-title"
  className="makeStyles-modal-1"
  closeAfterTransition={true}
  onClose={[Function]}
  open={true}
>
  <ForwardRef(Fade)
    in={true}
  >
    <div
      className="makeStyles-paper-2"
    >
      <h2
        id="spring-modal-title"
      >
        Warning
      </h2>
      <p
        id="spring-modal-description"
      >
        <b>
          Maximum limit reached
        </b>
      </p>
      <div
        className="makeStyles-footerButton-3"
      >
        <WithStyles(ForwardRef(Button))
          onClick={[Function]}
        >
          Ок
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-buttonLastModal-4"
          color="secondary"
          onClick={[Function]}
          variant="outlined"
        >
          <b>
            Удалить все избранное
          </b>
        </WithStyles(ForwardRef(Button))>
      </div>
    </div>
  </ForwardRef(Fade)>
</ForwardRef(Modal)>
`;
